Program.cs

```diff
@@ -852,6 +865,84 @@ public void CancelValidation()
             ValidatingDotnet = false;
             ValidatingJava = false;
         }
+
+        public async void CheckForUpdates()
+        {
+            var currentVersion = Assembly.GetEntryAssembly().GetName().Version;
+            Debug.WriteLine($"Currently running Hammer v{currentVersion}. Checking for updates...");
+
+            string tags = await GetRepoTags();
+            if (String.IsNullOrEmpty(tags)) {
+                return;
+            }
+
+            var latestVersion = GetLatestVersion(tags);
+
+            if (latestVersion == null) {
+                return;
+            }
+
+            if (latestVersion > currentVersion)

```

Comment:
Does this comparison work? From what I can see in https://docs.microsoft.com/en-us/dotnet/api/system.version?view=netframework-4.8#comparing-version-objects, you need to use the CompareTo method.

Reply:
Yes it seemed to in my testing... I got it from an SO answer, but I tested the different values. I'll ask a .NET expert.